libs/libc/stream/lib_bufferedoutstream.c

```diff
@@ -0,0 +1,154 @@
+/****************************************************************************
+ * libs/libc/stream/lib_bufferedoutstream.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <unistd.h>
+#include <assert.h>
+#include <errno.h>
+
+#include <nuttx/fs/fs.h>
+
+#include "libc.h"
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: bufferedoutstream_flush
+ ****************************************************************************/
+
+static int bufferedoutstream_flush(FAR struct lib_outstream_s *this)
+{
+  FAR struct lib_bufferedoutstream_s *rthis =
+    (FAR struct lib_bufferedoutstream_s *)this;
+  int ret = OK;
+  int i;
+
+  if (rthis->pending > 0)
+    {
+      /* Try to write full internal buffer to fd */
+
+      for (i = 0; i < rthis->pending; )
```

Comment:
don't need for loop, the low level wrapper should handle -EINTR for us

Reply:
OK